cosmwasm/contracts/crosschain-registry/src/execute.rs

```diff
@@ -40,6 +42,100 @@ pub struct ContractAliasInput {
     pub new_alias: Option<String>,
 }
 
+pub fn propose_pfm(
```

Comment:
wondering if I'm understading this correctly:

- the proposer can be anyone and basically trying to send coin cross-chain with PFM memo
- waiting for ack ensure that the sent packet is ack-ed -> `acknowledged = true`
- `ibc-hook` execute this contract with `validate_pfm` -> `validated = true`
  - `validate_pfm` also checks if sender is actually from `ibc-hook`and first originated from this contract
- these info lives in registry contract ensuring that target chain is pfm enabled

Reply:
correct! One caveat though... now that PFM enforces an intermediate sender... we will need to update this to validate that.